src/test/java/games/strategy/test/EqualsPredicate.java

```diff
@@ -0,0 +1,72 @@
+package games.strategy.test;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+
+@Immutable
+final class EqualsPredicate {
```

Comment:
This is effectively the implementation of `TestUtil#equals()`.  I modeled it as a Method Object because it is recursive and moderately complex.